Standards/scs-0200-v1-sonobuoy-kaas-conformance-tests.md

```diff
@@ -0,0 +1,168 @@
+
+
+#### _Option 2_ go approach [2]: Reuse the kubernetes own e2e test infrastructure and framework
+
+As mentioned above, Sonobuoy already provides [Kubernetes own e2e tests][k8s-e2e-tests] as a plugin.
+These tests contain a vast number of examples that we could reuse and adapt to our needs.
+For the implementation, we could reuse the e2e tests and the framework in an adapted structure.
+More precisely this would lead us to make use of the test framework [ginkgo][ginkgo].
+The entry point for the implementation would be the build process of the Dockerimage which contains
+the e2e tests. Therefore we could copy the setup of the build process from
+[kubernetes/test/conformance/image][conformance-image] and adapt it to our requirements.
+The build process requires some files from the kubernetes repository.
+More precisely from:
+  - [kubernetes/cluster](https://github.com/kubernetes/kubernetes/tree/master/cluster)
+  - [kubernetes/test/e2e/framework](https://github.com/kubernetes/kubernetes/tree/master/test/e2e)
+  - [kubernetes/test/conformance/image/go-runner](https://github.com/kubernetes/kubernetes/tree/master/test/conformance/image/go-runner)
+
+
+PROS:
+   - The [Kubernetes' own e2e tests][k8s-e2e-tests] already provide a vast amount of examples from which we can develop our own tests
+   - Compared to _option 1_, the [non-goals][e2e-frame-nongoals] of the [e2e-framework][e2e-frame] can be seen as the advantages of using [Kubernetes' own e2e-tests][k8s-e2e-tests].
+
+
+CONS:
+  - not easy to implement, as we would have to copy part of the Kubernetes repository and track the changes there as well.
+  - According to [README.md](https://github.com/kubernetes/kubernetes/tree/master/cluster#readme), part of it seems to be outdated and might change in the future.
+   - Compared to _option 1_, the [goals][e2e-frame-goals] of the [e2e-framework][e2e-frame] can be seen as the disadvantages of using [Kubernetes' own e2e-tests][k8s-e2e-tests].
+
+
+> TODO: provide proof of work: _kaas-sonobuoy-go-example-k8s-e2e_
+
+
+#### _Option 3_ Write Python scripts for tests
+
+Apply the already implemented conformance test python scripts and also continue
+writing future tests in python. Therefore, we would need to write a wrapper
+around the conformance tests scripts in order to make them executable as a
+Sonobuoy plug in. This includes  the collection of the test results and the
+Generation of the "done" file at the end. Furthermore, we could do this with a
+simple handler that executes each test script in a sequential order. All of
+this can then be stored in a container image and uploaded to a container
+registry that Sonobuoy can use within the k8s-cluster-api-provider.
+
+This approach also leaves the decision open as to which test framework should be
+used for Python. Hence, if we follow this approach, we need to create a
+framework of our own.
```

Comment:
```suggestion
Sonobuoy makes it possible to write tests in Python and execute like other tests in a pod on the K8s cluster.
It would therefore be possible to keep on writing conformance tests in Python.

This option would require a wrapper in order to make the tests scripts executable as Sonobuoy plugins.
This wrapper, as mentioned earlier, would need to include the collection of test results as well as the generation of the "Done" file after the test execution is finished. This could be managed by executing each test script in a sequential order. All of this could then be stored in a container image and uploaded to a registry in order to be usable by Sonobuoy within the k8s-cluster-api-provider.

This approach also leaves the decision open as to which test framework should be
used for Python, which should be decided in a secondary Decision Record.
```

Rewrote this part.